### PR TITLE
Fix bad cache setting

### PR DESCRIPTION
**Pull request by maschwenk**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/requirement.rb [hunk 1]
```diff
@@ -239,7 +239,10 @@ def expand(dependent, cache_key: nil, &block)
         end
       end
 
-      cache[cache_key][cache_id dependent] = reqs.dup if cache_key.present?
+      if cache_key.present?
+        cache[cache_key] ||= {}
```
Comment (MikeMcQuaid) [inline Library/Homebrew/requirement.rb:243]: This is already done on line 225

**Review by MikeMcQuaid — COMMENTED** [review]
Can you explain the motivation for this? Thanks!

**Comment by maschwenk** [comment]
@MikeMcQuaid will do, sorry opened as a draft just so I wouldn't forget it and was gonna come back with a more formal description/explanation

**Comment by MikeMcQuaid** [comment]
Closing until then. Shout when you're ready, thanks.

**Pull request closed without merging**